client/src/pages/Review/Review.tsx

```diff
@@ -58,86 +68,35 @@ const Review = () => {
     return () => window.removeEventListener("beforeunload", beforeUnload);
   }, []);
 
-  useEffect(() => {
-    if (currentCourseKey === null) {
-      setTimeout(() => history.pushState(null, "", "/app"), 10);
-    } else {
-      setTimeout(() => history.pushState(null, "", `/app/${currentCourseKey}`), 10);
-    }
-  }, [currentCourseKey]);
-
-  // Memorize the course key lookup in format of { [key]: course } for performance
-  const courseKeyLookup = useMemo(
-    () =>
-      Object.fromEntries(
-        courses.map((course) => [course.key, course])
-      ) as Record<string, Course>,
-    [courses]
-  );
-
-  // Memorize the course based on the course key
-  const currentCourse = useMemo(
-    () =>
-      currentCourseKey && currentCourseKey in courseKeyLookup
-        ? courseKeyLookup[currentCourseKey]
-        : null,
-    [currentCourseKey, courseKeyLookup[currentCourseKey || ""]]
-  );
```

Comment:
This lookup table is for `currentCourseKey` to quickly find its corresponding `Course` object. 
The table is in the format of `{ key: Course }`. To avoid iterating the `courses` array.

Reply:
Yes but in reality the probability that we have more than 10 courses is very very low, so I don't mind iterating over the array. O(n) effectively becomes O(1)

Reply:
This was introduced in #118 for improving performance issue (that I never really experience...), so yeah if you're good with removing this then I'm good too, we shall agree that it's the assessment list is _the_ dragon.

Reply:
Yeah I do believe there is an issue somewhere but not here